### PR TITLE
Task/migrate template gsi 390

### PR DESCRIPTION
**Pull request by galinatremper**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    "Programming Language :: Python :: 3.11",
+    "License :: OSI Approved :: Apache Software License",
+    "Topic :: Internet :: WWW/HTTP :: HTTP Servers",
+    "Topic :: Software Development :: Libraries",
+    "Intended Audience :: Developers",
+]
+dependencies = [
+    "typer >= 0.7.0",
+    "ghga-service-commons[api] >= 0.7.0",
+    "ghga-event-schemas >= 0.13.0",
```
Comment (TheByronHimes) [inline pyproject.toml]: Do you need all of these? For example, I couldn't find an instance of `ghga-event-schemas` being used in the project.

**Review by TheByronHimes — DISMISSED** [review]
Suggestion: consider removing any dependencies from pyproject.toml (and elsewhere) that are not being used. Looks good though

File: pyproject.toml (hunk 1 above)
Comment (galinatremper) [reply pyproject.toml]: You're right, I'll delete them.

**Review by KerstenBreuer — COMMENTED** [review]
I think for all CLI clients we should apply the same strategy as for libraries when it comes to dependency restrictions in the pyproject.toml:

Set lower bound to the first version that worked. Set upper bound to exclude versions that might introduce incompatibilities according to semver.

**Review by TheByronHimes — CHANGES_REQUESTED** [review]
Something has been mixed up: `requirements-dev.in` should only be a basic pip-requirements type file, not a lock file. I noticed `ghga-event-schemas` is still in your `requirements-dev.txt`. Once you fix the issue with the **.in** file, can you delete and rebuild the lock files?

**Review by TheByronHimes — DISMISSED** [review]

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    "License :: OSI Approved :: Apache Software License",
+    "Topic :: Internet :: WWW/HTTP :: HTTP Servers",
+    "Topic :: Software Development :: Libraries",
+    "Intended Audience :: Developers",
+]
+dependencies = [
+    "typer >= 0.7.0",
+    "pydantic >1.10.6, <2",
+    "linkml >= 1.5.7, <1.6.0",
+    "linkml-validator >= 0.4.5"
```
Comment (KerstenBreuer) [inline pyproject.toml]: Typer should also have an upper bound. Moreover, I pinned the linkml dependencies as they are very unreliable and we can only guarantee that thinks work for a very specific version of them:

```suggestion
    "typer ~= 0.7.0",
    "pydantic >1.10.6, <2",
    "linkml==1.6.1",
    "linkml-validator==0.4.5"
```

**Review by KerstenBreuer — CHANGES_REQUESTED** [review]

File: pyproject.toml (hunk 2 above)
Comment (lkuchenb) [reply pyproject.toml]: In metldata we have upgraded to the latest versions so we should use them in the tools, too
```
    "linkml==1.6.1",
    "linkml-runtime==1.6.0",
    "linkml-validator==0.4.5",
```

**Review by KerstenBreuer — APPROVED** [review]

**Review by TheByronHimes — APPROVED** [review]

**Pull request merged**